### PR TITLE
[FLINK-12461] Document Scala 2.12.x situation

### PR DESCRIPTION
**Pull request by aljoscha**
## What is the purpose of the change

Scala versions after 2.12.8 are not binary compatible with previous versions, we link to the relevant release notes and point users towards building their own release using a more recent version.

It's a pure documentation change, no testing needed.

### CODE REVIEW

File: docs/flinkDev/building.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ Since version 1.7 Flink builds with Scala version 2.11 (default) and 2.12.
 
 To build FLink against Scala 2.12, issue the following command:
 {% highlight bash %}
-mvn clean install -DskipTests -Dscala-2.12
+mvn clean install -DskipTests -Dscala-2.12 -Dscala.version=2.12.12
```
Comment (zentol) [inline docs/flinkDev/building.md:135]: Shouldn't we use the default scala version here?

File: docs/dev/project-configuration.md [hunk 2]
```diff
@@ -146,6 +146,23 @@ pick the Scala version that matches their application's Scala version.
 Please refer to the [build guide]({% link flinkDev/building.md %}#scala-versions)
 for details on how to build Flink for a specific Scala version.
 
+Scala versions after 2.12.8 are not binary compatible with previous 2.12.x
+versions. The Flink project does therefore not offer releases for versions past
+2.12.8. If you want, you can build Flink yourself for a later Scala version.
+Refer to the [build guide]({%link flinkDev/building.md %}#scala-versions) for
+more information, as mentioned above. For this to work, you need to add
+`-Djapicmp.skip` to skip binary compatibility checks when building.
```
Comment (sjwiesman) [inline docs/dev/project-configuration.md:154]: It was a bit wordy, I think this sounds a little clearer. I also want to make it very clear this is out of our hands.

```suggestion
Scala versions after 2.12.8 are not binary compatible with previous 2.12.x
versions, preventing the Flink project from upgrading its 2.12.x builds beyond 2.12.8.
Users can build Flink locally for latter Scala versions by following the above mentioned
[build guide]({% link flinkDev/building.md %}#scala-versions). For this to work, users
need to add `-Djapicmp.skip` to skip binary compatibility checks when building.
```

**Review by sjwiesman — APPROVED** [review]
Thanks for taking the time, I made a suggestion to be a little more clear with our language but the basic content makes sense. Also +1 to Chesnay's comment. Feel free to address when merging :)

File: docs/flinkDev/building.md (hunk 1 above)
Comment (aljoscha) [reply docs/flinkDev/building.md:135]: This is to show that you can use a custom version, which is the purpose of this PR. I think you only want to build it yourself to change the exact Scala binary version. Otherwise, you can just use the binary releases.

Comment (sjwiesman) [reply docs/flinkDev/building.md:135]: Maybe put <scala_version>?

Comment (aljoscha) [reply docs/flinkDev/building.md:135]: I'll add a second example for building against a specific version and will use `<scala version>`

**Comment by aljoscha** [comment]
Thanks for the reviews!
